src/mon/MgrMonitor.cc

```diff
@@ -84,8 +84,9 @@ void MgrMonitor::update_from_paxos(bool *need_bootstrap)
     check_subs();
 
     if (version == 1
-	|| (map.get_available()
-	    && (!old_available || old_gid != map.get_active_gid()))) {
+        || command_descs.empty()
```

Comment:
if there is no active mgr, why shall we load the cmd desc for the mon client and redirect it to a nonexistent mgr? the net result is that it will be waiting for an active mgr in mgrmap. and strict speaking, the command_desc could be stale once the new mgr is activated, as it will send a beacon to mon and update it with the commands it will be able to serve. and the "commands" here could be different from the ones stored in monstore.

Reply:
This patch relies on the assumption that a client will always block if there is no active mgr. That was a question I posed on ceph-devel before working on the patch, and Sage's reply was that the client is supposed to hang.
Given it's the monitor's responsibility to tag (by means of flags) which commands are the mgr's, we need to have a list of mgr commands that can be relayed to the client -- even if that means to relay a stale list, given we know of no active mgr's command list, so that the client can hang waiting for an active mgr.
Now, if it is established that the client is not supposed to hang, instead getting an EINVAL because the monitor does not handle the command, nor does it know of any manager that will, then a different patch needs to be worked on.

Reply:
> the client is supposed to hang.

ahh, i am good then.